### ISSUE TITLE
space-after-type-colon should accept newlines

### ISSUE THREAD

**Issue by turadg** [gajus/eslint-plugin-flowtype#31]
With config,

```
    "flowtype/space-after-type-colon": [2, "always"],
```

Code like this fails,

```
const Fn = ({className}:
  {className?: string}) => {
```

There's no space directly after the colon, but a newline should count to avoid trailing whitespace.

**Comment by ffxsam** [comment]
How funny, I just came here to start a discussion about the very same thing. 100% agree.

**Comment by gajus** [comment]
@turadg are you available to raise a PR for this? It should be fairly trivial change. I'd like to see it as a configuration, though – `allowLineBreak`.

**Comment by turadg** [comment]
> I'd like to see it as a configuration, though – allowLineBreak.

In what cases would someone not want to allow a line break?

**Comment by gajus** [comment]
> In what cases would someone not want to allow a line break?

This is a code style question. Describing a "case" would be rather difficult. Personally, I have never needed to use a newline character when describing function types. That said, I don't use inline type declarations either.

**Comment by turadg** [comment]
> Personally, I have never needed to use a newline character when describing function types.

That's cool. I'm just trying to understand why someone would want a line break to error when they want to enforce a space after the colon.

If I submit a PR, it won't have the `allowLineBreak` option. Would a PR without that be accepted?

**Comment by gajus** [comment]
> If I submit a PR, it won't have the allowLineBreak option. Would a PR without that be accepted?

Go ahead.

Thank you

**Comment by ffxsam** [comment]
>Personally, I have never needed to use a newline character when describing function types

That was before Flow and TypeScript. :)

**Comment by gajus** [comment]
> Same here. But that was before Flow and TypeScript. 😄

I think I didn't make myself clear. I have meant to say, that I am only using named type bindings, e.g.

```js
type TypeName = string;

(name: TypeName) => {};
```

(This example is too simplistic. This is useful only when dealing with non-primitive types.)

As a result, there is never a case when I would need to have type declaration on a different line than the parameter itself is.

**Comment by ffxsam** [comment]
This is the case for wanting ESLint to accept newlines in Flow annotations:

```js
export function someFunc(name: TypeName, someArg: ?string, finalArg: Object):
boolean {
  // ...
}
```

If the function declaration were not wrapped, ` boolean {` would exceed the 80-char-per-line limit.

**Comment by gajus** [comment]
> If the function declaration were not wrapped, boolean { would exceed the 80-char-per-line limit.

Hard-wrapping is 80s. :-)

All modern text editors can do soft-wrapping, while IDEs take a notch further and use intelligent soft-wrapping that respects common syntax boundaries.

Regardless, discussion about the pros and cons of hard-wrapping vs soft is outside the scope of this issue thread. I am happy to support the feature if it is a requirement.

**Comment by turadg** [comment]
@gajus great point! I'm just going to switch to soft wrapping. 😜
